File: generated-code/typescript/codecraft/TcpReadWrite/src/vec2-int.ts
import { Stream } from "./stream";

/**
 * 2 dimensional vector.
 */
export class Vec2Int {
    /**
     * `x` coordinate of the vector
     */
    x: number
    /**
     * `y` coordinate of the vector
     */
    y: number

    constructor(x: number, y: number) {
        this.x = x;
        this.y = y;
    }

    /**
     * Read Vec2Int from input stream
     */
    static async readFrom(stream: Stream): Promise<Vec2Int> {
        let x;
        x = await stream.readInt();
        let y;
        y = await stream.readInt();
        return new Vec2Int(x, y)
    }

    /**
     * Write Vec2Int to output stream
     */
    async writeTo(stream: Stream) {
        let x = this.x;
        await stream.writeInt(x);
        let y = this.y;
        await stream.writeInt(y);
    }
}

File: generated-code/python/codecraft/FileReadWrite/vec2_int.py
from stream_wrapper import StreamWrapper

class Vec2Int:
    """2 dimensional vector."""

    __slots__ = ("x","y",)

    x: int
    y: int

    def __init__(self, x: int, y: int):
        self.x = x
        """`x` coordinate of the vector"""
        self.y = y
        """`y` coordinate of the vector"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Vec2Int":
        """Read Vec2Int from input stream
        """
        x = stream.read_int()
        y = stream.read_int()
        return Vec2Int(x, y)
    
    def write_to(self, stream: StreamWrapper):
        """Write Vec2Int to output stream
        """
        stream.write_int(self.x)
        stream.write_int(self.y)
    
    def __repr__(self):
        return "Vec2Int(" + \
            repr(self.x) + \
            ", " + \
            repr(self.y) + \
            ")"